scene/gui/popup_menu.cpp

```diff
@@ -174,7 +174,12 @@ void PopupMenu::_activate_submenu(int over) {
 		if (pum->get_item_count() > 0 && Input::get_singleton()->is_action_just_pressed("ui_accept")) {
 			pum->set_current_index(0);
 		}
+	}
+
+	// Must be in the middle, see GH-58126.
+	pm->popup();
 
+	if (pum) {
```

Comment:
I'd explain instead why setting up autohide areas must be done after `popup()`:
```suggestion
	pm->popup();

	// Setting autohide areas must be done after `popup()` which can move the popup (to fit it into the viewport).
	if (pum) {
```

Actually this made me wonder: does `pm->popup();` need to be in the middle? Can't it be before `pum->set_current_index(0);`?

Reply:
Commented updated.

> Actually this made me wonder: does pm->popup(); need to be in the middle? Can't it be before `pum->set_current_index(0);`?

Yep, seems to work fine doing that also. I wonder why I changed the position before.

Reply:
>Yep, seems to work fine doing that also. I wonder why I changed the position before.

Also not sure what would be the reason, looks fine as is.
But in case of a regression let's remember to add a comment explaining the reason. :wink: